docs/src/design/aux_table/main.md

```diff
@@ -1,3 +1,184 @@
 # Auxiliary Table
 
-TODO
+The Auxiliary Table reduces the number of columns required by the execution trace by stacking the execution traces of 4 co-processors that are expected to generate significantly fewer rows than the other VM processors (the decoder, stack, and range checker).
+
+## Auxiliary Table segments
+
+The co-processors in the auxiliary table are:
+
+- [Hash Processor](./hasher.md) (17 columns; degree 8)
+- Bitwise & Power of Two Processor, which combines 2 co-processors:
+  - [Bitwise Processor](./bitwise.md) (13 columns; degree 6)
+  - [Power of Two Processor](./pow2.md) (12 columns; degree 3)
+- [Memory Processor](./memory.md) (14 columns; degree 6)
+
+Each co-processor is identified by a set of selector columns which identify its segment in the auxiliary table and cause its constraints to be selectively applied.
+
+During the finalization of the overall execution trace, each co-processor's trace and selector columns are appended to the Auxiliary Table one after another, such that when one co-processor trace ends the trace of the next co-processor starts in the subsequent row.
+
+Additionally, a padding segment must be added to the Auxiliary Table so that the number of rows in the table always matches the overall trace length of the other VM processors, regardless of the length of the co-processor traces. The padding will simply contain zeroes.
+
+## Design Requirements
+
+- The minimum width of the Auxiliary Table is 18 columns, which will fit the Hash processor and 1 selector column to select it.
+- The maximum number of constraint degrees for the VM is 9.
+
+### Hasher
+
+- The hasher's constraints are already degree 8, so we are restricted to a 1-degree selector flag.
+- The hasher requires 17 columns, so its selector flag must require a single column to keep our aux table to the minimum.
+- Each hash operation in the hash processor is performed in an 8-row cycle that must begin at a row number equal to $0 mod8$, so the hash processor's trace must begin on a row number equal to $0 mod8$.
```

Comment:
nit: I'm not sure if "$0 mod8$" will render correctly. should it be `$0 \mod 8$`? or maybe `$0\ mod\ 8$`?

Reply:
Updated. I think it doesn't look great either way, actually, but this is more correct.